src/js/components/Button/StyledButtonKind.js

```diff
@@ -30,7 +30,7 @@ const radiusStyle = (props) => {
 
 const fontStyle = (props) => {
   const size = props.sizeProp || 'medium';
-  const data = props.theme.text[size];
+  const data = props.theme.text[props.kind?.font?.size || size];
```

Comment:
Button index.d.ts has `kind?: string;`, which doesn't align with the property being an object.

Reply:
Also, shouldn't the size be coming from the theme instead of the property? Should it be something like `props.theme.text[props.theme?.[props.kind]?.font?.size`?

Reply:
Removing this adjustment, it turns out the font size for the label was already supported so this change was not driving the styling.